src/svg_to_png/lib/ThreatModel.py

```diff
@@ -0,0 +1,218 @@
+import xml.etree.ElementTree as ET
+from xml.etree.ElementTree import Element
+import re
+
+from .FreeTextAnnotation import FreeTextAnnotation
+from .GenericDataFlow import GenericDataFlow
+from .GenericProcess import GenericProcess
+from .GenericTrustBorderBoundary import GenericTrustBorderBoundary
+from .GenericTrustLineBoundary import GenericTrustLineBoundary
+from .GenericDataStore import GenericDataStore
+from .GenericExternalInteractor import GenericExternalInteractor
+
+def build_tag(schema, tag):
+    return f'{schema}{tag}'
+
+
+def print_all(iter):
+    for elem in iter:
+        print(elem.tag, elem.attrib)
+        
+THREAT_MODELING_XMLNS = "{http://schemas.datacontract.org/2004/07/ThreatModeling.Model}"
+ABSTRACTS_XMLNS = "{http://schemas.datacontract.org/2004/07/ThreatModeling.Model.Abstracts}"
+ARRAY_XMLNS = "{http://schemas.microsoft.com/2003/10/Serialization/Arrays}"
+KNOWLEDGE_BASE_XMLNS = "{http://schemas.datacontract.org/2004/07/ThreatModeling.KnowledgeBase}"
+        
+def get_shape_details(shape):
+    height = shape.find(build_tag(ABSTRACTS_XMLNS, "Height")).text
+    width = shape.find(build_tag(ABSTRACTS_XMLNS, "Width")).text
+    left = shape.find(build_tag(ABSTRACTS_XMLNS, "Left")).text
+    top = shape.find(build_tag(ABSTRACTS_XMLNS, "Top")).text
+    
+    return height, width, left, top
+
+def get_curve_details(curve):
+    handleX = curve.find(build_tag(ABSTRACTS_XMLNS, "HandleX")).text
+    handleY = curve.find(build_tag(ABSTRACTS_XMLNS, "HandleY")).text
+    sourceX = curve.find(build_tag(ABSTRACTS_XMLNS, "SourceX")).text
+    sourceY = curve.find(build_tag(ABSTRACTS_XMLNS, "SourceY")).text
+    targetX = curve.find(build_tag(ABSTRACTS_XMLNS, "TargetX")).text
+    targetY = curve.find(build_tag(ABSTRACTS_XMLNS, "TargetY")).text
+    
+    return handleX, handleY, sourceX, sourceY, targetX, targetY
+
+def get_element_name(shape):
+    properties = shape.find(build_tag(ABSTRACTS_XMLNS, "Properties"))
+    any_type_properties = properties.findall(build_tag(ARRAY_XMLNS, "anyType"))
+    name = any_type_properties[1][2].text
+    return name if name else ""
+
+def is_in(candidate, boundary):
+    return boundary.left <= candidate.left and boundary.top <= candidate.top and boundary.left + boundary.width >= candidate.left + candidate.width and boundary.top + boundary.height >= candidate.top + candidate.height
+
+def set_appropriate_groups(candidate, boundary):
+    if is_in(candidate, boundary):
```

Comment:
Is this more of a tree of groups? Node is in bounday X, which is in boundary Y?

Reply:
Right, each node is in one boundary and each boundary can be in a boundary. So from a boundary perspective, it's a tree, but we are really only interested in the node perspective.

Reply:
Gotcha. yeah i think it would be helpful to get privacy champs perspective on what is useful here. But easy to update